src/globus_sdk/exc/err_info.py

```diff
@@ -64,9 +64,15 @@ def __init__(self, error_data: t.Dict[str, t.Any]):
         self.session_required_single_domain = t.cast(
             t.Optional[t.List[str]], data.get("session_required_single_domain")
         )
-        self.session_required_policies = t.cast(
-            t.Optional[str], data.get("session_required_policies")
-        )
+
+        # get str|None and parse as appropriate
+        session_required_policies = data.get("session_required_policies")
+        if isinstance(session_required_policies, str):
+            self.session_required_policies: t.Optional[
```

Comment:
Not a big deal, but it feels odd to only have the type information in the if branch and not the else branch for some reason. Does a pattern like this seem better or is it just me?

```
self.foo: t.Optional[t.fooType] = None
if condition_for_foo_existing:
    self.foo = bar
```

Reply:
It's stopped looking odd to me to do the typing assignment like this, but that doesn't mean it isn't odd. It just means that I've done it too many times. 😉 

I think I'll make a modification here to be similar to your suggestion, but also add a warning log for the present-but-not-string case. (I don't think we can do more than that.)